mobile-app/src/components/HeaderOptions.tsx

```diff
@@ -73,7 +75,8 @@ export function viewHabitOptions(
 export function viewProfileOptions(
   colorScheme: string,
 ): NativeStackNavigationOptions {
-  const { username } = useGlobalSearchParams<{ username: string }>();
+  const { theirUserId } = useGlobalSearchParams<{ theirUserId: string }>();
+  const username = useAtomValue(friendUsernameAtom(theirUserId as string));
```

Comment:
I think this will cause a problem if you search someone up and view their profile but they aren't your friend.

Maybe see the userinfo atom:

```ts
export const getUserInfoAtom = atomFamily((userId: string) =>
  atom(async (get) => fetchUserInfo({ userId })),
);
```

And add a `getUserUsernameAtom` atom underneath